Adds documentation to `substr` function to cover when `length` exceeds input length (#30483)
hashicorp/terraform: website/docs/language/functions/substr.mdx

> substr("hello world", -5, -1)
world
```

> The length may be greater than the length of the string, in which case the substring
> will be the length of all remaining characters.

Reviewer: ```suggestion
If the length is greater than the length of the string, the substring
will be the length of all remaining characters.
```

Author: Looks good!
